ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -11086,6 +11171,9 @@ ABoundActivity * HqlCppTranslator::doBuildActivityOutput(BuildCtx & ctx, IHqlExp
 
     buildInstancePrefix(instance);
 
+    if (!genericFileTypeFormat.isEmpty())
+        instance->startctx.addQuotedF("virtual const char * queryFormat() { return \"%s\"; }", genericFileTypeFormat.str());
```

Comment:
genericFileTypeFormat not initialised if the global option has been used to enable generic writes.  Can probably default it to str(queryChild(2)->queryName())

Reply:
Now uses a generic file type format if provided, expr->queryChild(2)->queryName() otherwise.

Reply:
This turned out to cause problems when compiling Roxie code; removed.